### DISCUSSION TITLE
UTC time?

### DISCUSSION THREAD

**Discussion by gerbert** [foldynl/QLog#105] [Ideas]
@foldynl,
Hi! Out of curiosity, do you consider adding UTC clock? E.g., like this:

```
        12:00:00
UTC: 10:00:00 01.01.1970
```
UTC text size will be 12pt.

Technically my question already has an implementation so I could put it in a branch and create a PR so you could take a look and make a decision...

Thanks!

**Comment by foldynl** [comment]
I'm a bit confused. Can you post a description of where the clock should be added? There is a Clock Widget then a localtime info for particular QSO partner. Where do you want to add a clock? All QLog's clock are in UTC

![image](https://user-images.githubusercontent.com/942769/179560754-b47f6fb5-b726-4083-b4c6-3aae2a3c660c.png)

 except following clock - this clock info is in a real localtime zone.

![image](https://user-images.githubusercontent.com/942769/179560848-316da2e4-0794-4bd4-b882-3eed44c1f03a.png)

**Comment by gerbert** [comment]
Well, I was thinking about something like this:
<img width="258" alt="local_utc" src="https://user-images.githubusercontent.com/3600366/179562610-0a4643e1-211c-4afb-9b36-08f2d521dc21.png">

**Comment by foldynl** [comment]
I saw your PR. Unfortunately, I breaks the design of the widget. The widget has to be simple. Unfortunately, you are duplicating information that are already shown.  

The primary aim of the widget is to show a clock in UTC. The local time is not important.

In general, QLog shows all times in UTC (with one exception which is present in my comment above).

why do you think that the PR should be applied? What is the use-case?

**Comment by gerbert** [comment]
The PR is different to this. This is just an idea. It's not yet even pushed anywhere.

The PR is about code optimization.

**Comment by foldynl** [comment]
and I forgot. All time has to reflect locale (date/time format) without timezone. It is important  emphasize "without timezone". QT  is curious that the Long Local time format is platform-dependent - I don't know why.

**Reply by gerbert** [reply]
That's why I used

```
QDateTime::currentDateTime().toLocalTime().toString("HH:mm:ss");
```

It extracts local time considering format given in toString() call. Technically, the format in this case can be adapted (if considering such an option) by adding additional parameters to HH:mm:ss... However now I'm thinking your approach has sence if the time format is always set by system settings...

**Comment by foldynl** [comment]
You are right. Now I am mixing two thinks together. Sorry...getting back to UTC. What is the use-case? Why do you want to change it?

**Reply by gerbert** [reply]
No problem. My point was that it could be useful to have UTC time displayed as well. Of course it won't replace big wall clock for local/UTC time, still will give a hint if HAM activities are bound to UTC time...
